### PR TITLE
Add schema validation

### PR DESCRIPTION
**Pull request by Rihyx**
Changes:
- add schema validation for projects
- add GitHub workflow
- bun init
- fix incorrect projects
- add .gitignore

### CODE REVIEW

File: scripts/projectValidation/projectSchema.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+});
+
+const logoSchema = z.object({
+  small: imageSchema.optional(),
+  large: imageSchema.optional(),
+  full: imageSchema.optional(),
+});
+
+const contractSchema = z.object({
+  contract: z.string(),
```
Comment (ekenigs) [inline scripts/projectValidation/projectSchema.ts:57]: Validate that it's proper EVM address?
```typescript
export const evmAddressSchema = z.string().regex(/^0x[a-fA-F0-9]{40}$/, {
  message: 'Invalid address. Must be 0x-prefixed 20 bytes hex string.',
});

```

File: scripts/projectValidation/projectSchema.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  .regex(/^[a-zA-Z0-9-_.]+$/)
+  .min(2)
+  .max(100);
+
+const chainEnumSchema = z.enum(["moonriver", "moonbeam"]);
+
+const defaultStringSChema = z.string().min(2).max(100);
+
+const imageSchema = z.object({
+  fileName: z.union(
```
Comment (ekenigs) [inline scripts/projectValidation/projectSchema.ts:40]: Do you want to be a bit more specific? Like line length and without special characters?

File: scripts/projectValidation/projectSchema.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    .string()
+    .url()
+    // ? for now we are allowing all medium links, has to be decided
+    // .regex(/^https:\/\/.*\.medium\.com\//)
+    .optional()
+    .or(z.string().url().startsWith("https://medium.com/").optional()),
+  telegram: z
+    .union([
+      z.string().url().startsWith("https://t.me/"),
+      z.string().url().startsWith("https://telegram.me/dexscreenerchat"),
```
Comment (ekenigs) [inline scripts/projectValidation/projectSchema.ts:81]:
```suggestion
      z.string().url().startsWith("https://telegram.me/"),
```

File: scripts/projectValidation/projectSchema.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    screenshots: logoSchema.array().optional(),
+    projectCreationDate: z
+      .number()
+      .min(946684800000)
+      .max(2524608000000)
+      .optional(),
+  })
+  .strict();
+
+export function validateProject(project: any) {
```
Comment (ekenigs) [inline scripts/projectValidation/projectSchema.ts:116]:
```suggestion
export function validateProject(project: unknown) {
```

**Review by ekenigs — CHANGES_REQUESTED** [review]

File: scripts/projectValidation/projectSchema.ts (hunk 1 above)
Comment (Rihyx) [reply scripts/projectValidation/projectSchema.ts:57]: Added

**Review by ekenigs — APPROVED** [review]

**Pull request merged**